File: DownloadFiles/TGNet/Chapter05/VB/Exercise5-1/AutoTester/AutoTester/Form1.vb
Public Class Form1
    Public Sub StartProgram(ByVal strAppname As String)
        'This routine will start a new process with the name and 
        'path contained in strAppname.        
        'TODO If a log file does not exist it will create one

        Process.Start(strAppname)

        'TODO Add code to log start time.
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnStartApp.Click
        Try
            StartProgram(txtAppname.Text)
        Catch ex As Exception
            MessageBox.Show(ex.Message.ToString)
        End Try
    End Sub

End Class

File: DownloadFiles/TGNet/Chapter05/C#/Exercise5-1/AutoTester/AutoTester/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;

namespace AutoTester
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        public void StartProgram(string strAppname)
        {
            Process.Start(strAppname);
        }


        private void btnStartApp_Click(object sender, EventArgs e)
        {
            try
            {
                StartProgram(txtAppName.Text);
            }
            catch (Exception ex)
            {

                MessageBox.Show(ex.Message.ToString());
            }
        }

      }
    }
